### PR TITLE
feat(android): add Amplify Android Developer Preview

### PR DESCRIPTION
**Pull request by div5yesh**
_Issue #, if available:_

_Description of changes:_

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/directory/directory.js [hunk 1]
```diff
@@ -6,13 +6,18 @@ const directory = {
...
           {
             title: "Getting started",
             route: "/lib/devpreview/getting-started",
             filters: ["ios"],
           },
+          {
+            title: "Getting started",
+            route: "/lib/devpreview/getting-started",
+            filters: ["android"],
+          },
```
Comment (eeatonaws) [inline src/directory/directory.js:20]: This extra "Getting started" block can be deleted and "android" can be added to the `filters` list on line 14.

File: src/fragments/lib/android.mdx [hunk 2]
```diff
@@ -12,4 +12,18 @@ This guide shows how to build an app using our Amplify Libraries for Android and
 
 <InternalLinkButton href="/lib/project-setup/prereq">
   <span slot="text">Get Started 🚀</span>
+</InternalLinkButton>
+
+## New! Amplify Android (Developer Preview)
+
+import ios0 from "/src/fragments/lib/devpreview/ios/getting_started/10_intro.mdx";
+
+<Fragments fragments={{ios: ios0}} />
```
Comment (eeatonaws) [inline src/fragments/lib/android.mdx:21]: Why is an iOS fragment being imported on an Android-specific page?

File: src/fragments/lib/devpreview/android/getting_started/10_intro.mdx [hunk 3]
```diff
@@ -0,0 +1,7 @@
+The Developer Preview of the Amplify Android Library is now exclusively using Kotlin and provides developers the ability to add cloud-based Auth, Storage, Data, and APIs to their apps. With this version, developers will be able to debug and contribute to the underlying open-source codebase completely in Kotlin.
```
Comment (eeatonaws) [inline src/fragments/lib/devpreview/android/getting_started/10_intro.mdx:1]:
```suggestion
The Developer Preview of the Amplify Android Library is now exclusively using Kotlin and provides developers the ability to add cloud-based Auth, Storage, DataStore, and APIs to their apps. With this version, developers will be able to debug and contribute to the underlying open-source codebase completely in Kotlin.
```

File: src/directory/directory.js [hunk 4]
```diff
@@ -6,13 +6,18 @@ const directory = {
     },
     items: {
       devpreview: {
-        title: "New! Amplify iOS (Developer Preview)",
+        title: "New! Amplify (Developer Preview)",
```
Comment (tjleing) [inline src/directory/directory.js:9]: Should this be
```suggestion
        title: "New! Amplify Mobile (Developer Preview)",
```
?

**Review by tjleing — APPROVED** [review]

File: src/fragments/lib/devpreview/android/getting_started/10_intro.mdx [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+
+This developer preview version of Amplify Android is layered on the [AWS SDK for Kotlin](https://aws.amazon.com/sdk-for-kotlin), which was released as Developer Preview last year. This allows for access to the AWS SDK for Kotlin for a breadth of service-centric APIs.
+
+You can also quickly get started by using our [Photo Sharing Sample App](https://github.com/aws-amplify/amplify-android-samples/tree/dev-preview).
```
Comment (eeatonaws) [inline src/fragments/lib/devpreview/android/getting_started/10_intro.mdx:5]: The current link goes to 404 page not found.
```suggestion
You can also quickly get started by using our [Photo Sharing Sample App](https://github.com/aws-amplify/amplify-android-samples/tree/main/PhotoSharing).
```

**Review by eeatonaws — CHANGES_REQUESTED** [review]

**Review by eeatonaws — APPROVED** [review]

**Pull request merged**